wordpress-mobile/WordPress-iOS
# Migrates PostListFilter to Swift. Fixes leaks.

Fixes #5624.

PostListFilter was leaking memory.  I spent a good amount of time tracking down the issue, and came down to the conclusion that it could be related with the heavy ObjC <-> Swift bridging going on.

To test:
- Launch the app
- Go to the list of posts / pages for a blog.
- Filter by published, draft, scheduled, trash.
- Make sure there's a post in each section and that it shows up properly.

To see the leaks:
- Launch the app with Instruments (run the Leaks instrument)
- Open the list of posts and pages for a blog.
- Simply wait until the leaks start popping up in the leaks view.
- Do the same in this PR to check the leaks are gone.

Needs review: @aerych
